app/containers/TracksContainer/TrackDetailsContainer/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ *
+ * TrackDetailsContainer Container
+ *
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { useParams } from 'react-router-dom';
+import { selectTrackDetails } from '../selectors';
+import { tracksContainerCreators } from '../reducer';
+import { TrackComponent } from '@app/components/TrackComponent/index';
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 2rem 0;
+`;
+
+const TrackDetailsWrapper = styled.div`
+  padding: 2rem;
+`;
+
+export function TrackDetailsContainer({ dispatchRequestTrackDetails, trackDetails }) {
+  const { trackId } = useParams();
+
+  useEffect(() => {
+    dispatchRequestTrackDetails(parseInt(trackId));
+  }, [trackId]);
+
+  return (
+    <Wrapper>
+      <TrackDetailsWrapper>
+        <TrackComponent
+          collectionName={trackDetails?.collectionName}
+          trackName={trackDetails?.trackName}
+          imageUrl={trackDetails?.artworkUrl100}
+          trackUrl={trackDetails?.previewUrl}
+          artistName={trackDetails?.artistName}
+          country={trackDetails?.country}
+          primaryGenreName={trackDetails?.primaryGenreName}
+          kind={trackDetails?.kind}
+          wrapperType={trackDetails?.wrapperType}
+          trackTimeMillis={trackDetails?.trackTimeMillis}
+        />
```

Comment:
only if trackDetail is present you can render this

Reply:
Updated accordingly